improver_tests/ensemble_copula_coupling/ensemble_copula_coupling/test_ConvertLocationAndScaleParametersToPercentiles.py

```diff
@@ -84,29 +99,79 @@ def test_check_data(self):
         variance. The resulting data values are the percentiles, which have
         been generated.
         """
-        data = np.array([[[[225.568115, 236.818115, 248.068115],
-                           [259.318115, 270.568115, 281.818115],
-                           [293.068115, 304.318115, 315.568115]]],
-                         [[[229.483322, 240.733322, 251.983322],
-                           [263.233307, 274.483307, 285.733307],
-                           [296.983307, 308.233307, 319.483307]]],
-                         [[[233.398529, 244.648529, 255.898529],
-                           [267.148499, 278.398499, 289.648499],
-                           [300.898499, 312.148499, 323.398499]]]],
-                        dtype=np.float32)
-
-        cube = self.current_temperature_forecast_cube
-        current_forecast_predictor = cube.collapsed(
-            "realization", iris.analysis.MEAN)
-        current_forecast_variance = cube.collapsed(
-            "realization", iris.analysis.VARIANCE)
-        percentiles = [10, 50, 90]
         plugin = Plugin()
         result = plugin._location_and_scale_parameters_to_percentiles(
-            current_forecast_predictor, current_forecast_variance, cube,
-            percentiles)
+            self.location_parameter, self.scale_parameter,
+            self.temperature_cube, self.percentiles)
         self.assertIsInstance(result, Cube)
-        self.assertArrayAlmostEqual(result.data, data)
+        np.testing.assert_allclose(
+            result.data, self.data, rtol=1.e-4)
+
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate."])
+    def test_masked_location_parameter(self):
+        """
+        Test that the plugin returns the correctly masked data when
+        given a location parameter that is masked.
+        """
+        mask = np.array([[1, 0, 0], [0, 0, 0], [0, 1, 0]])
+        expected_mask = np.array([[[1, 0, 0], [0, 0, 0], [0, 1, 0]],
+                                  [[1, 0, 0], [0, 0, 0], [0, 1, 0]],
+                                  [[1, 0, 0], [0, 0, 0], [0, 1, 0]]])
```

Comment:
See comment on test_ConvertLocationAndScaleParametersToProbabilities.py
For this file's worth.